src/lib/services/api-token-service.ts

```diff
@@ -142,9 +160,23 @@ export class ApiTokenService {
         }
     }
 
-    private generateSecretKey({ project, environment }) {
+    private findInvalidProject(errorDetails, projects) {
+        if (!errorDetails) {
+            return 'invalid';
+        }
+        let invalidProject = projects.find((project) => {
+            return errorDetails.includes(`=(${project})`);
+        });
+        return invalidProject || 'invalid';
+    }
+
+    private generateSecretKey({ projects, environment }) {
         const randomStr = crypto.randomBytes(28).toString('hex');
-        return `${project}:${environment}.${randomStr}`;
+        if (projects.length > 1) {
+            return `[]:${environment}.${randomStr}`;
```

Comment:
So multi project tokens now gets resolved from the db, fair enough :)

Reply:
As opposed to not being supported at all, don't get me wrong :)

Reply:
Yeah, it's a big tradeoff here. All the other token types can be determined just from their secret signature, this is the first time they're no longer stateful. Means we can't yeet them from Unleash instance to Unleash instance unfortunately but it keeps the token small